diff --git a/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs b/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs
index ef6e60220..4cf1ac0f8 100644
--- a/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs
+++ b/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs
@@ -740,13 +740,14 @@ namespace MediaBrowser.Controller.MediaEncoding
             var isWindows = OperatingSystem.IsWindows();
             var isLinux = OperatingSystem.IsLinux();
             var isMacOS = OperatingSystem.IsMacOS();
+            var isFreeBSD = OperatingSystem.IsFreeBSD();
             var optHwaccelType = options.HardwareAccelerationType;
             var vidDecoder = GetHardwareVideoDecoder(state, options) ?? string.Empty;
             var isHwTonemapAvailable = IsHwTonemapAvailable(state, options);
 
             if (string.Equals(optHwaccelType, "vaapi", StringComparison.OrdinalIgnoreCase))
             {
-                if (!isLinux || !_mediaEncoder.SupportsHwaccel("vaapi"))
+                if ((!isLinux || !isFreeBSD) || !_mediaEncoder.SupportsHwaccel("vaapi"))
                 {
                     return string.Empty;
                 }
@@ -3859,10 +3860,11 @@ namespace MediaBrowser.Controller.MediaEncoding
             }
 
             var isLinux = OperatingSystem.IsLinux();
+            var isFreeBSD = OperatingSystem.IsFreeBSD();
             var vidDecoder = GetHardwareVideoDecoder(state, options) ?? string.Empty;
             var isSwDecoder = string.IsNullOrEmpty(vidDecoder);
             var isSwEncoder = !vidEncoder.Contains("vaapi", StringComparison.OrdinalIgnoreCase);
-            var isVaapiOclSupported = isLinux && IsVaapiSupported(state) && IsVaapiFullSupported() && IsOpenclFullSupported();
+            var isVaapiOclSupported = (isLinux || isFreeBSD) && IsVaapiSupported(state) && IsVaapiFullSupported() && IsOpenclFullSupported();
 
             // legacy vaapi pipeline(copy-back)
             if ((isSwDecoder && isSwEncoder)
@@ -4618,8 +4620,9 @@ namespace MediaBrowser.Controller.MediaEncoding
             var isWindows = OperatingSystem.IsWindows();
             var isLinux = OperatingSystem.IsLinux();
             var isMacOS = OperatingSystem.IsMacOS();
+            var isFreeBSD = OperatingSystem.IsFreeBSD();
             var isD3d11Supported = isWindows && _mediaEncoder.SupportsHwaccel("d3d11va");
-            var isVaapiSupported = isLinux && IsVaapiSupported(state);
+            var isVaapiSupported = (isLinux || isFreeBSD) && IsVaapiSupported(state);
             var isCudaSupported = (isLinux || isWindows) && IsCudaFullSupported();
             var isQsvSupported = (isLinux || isWindows) && _mediaEncoder.SupportsHwaccel("qsv");
             var isVideotoolboxSupported = isMacOS && _mediaEncoder.SupportsHwaccel("videotoolbox");
@@ -4925,7 +4928,7 @@ namespace MediaBrowser.Controller.MediaEncoding
 
         public string GetVaapiVidDecoder(EncodingJobInfo state, EncodingOptions options, MediaStream videoStream, int bitDepth)
         {
-            if (!OperatingSystem.IsLinux()
+            if ((!OperatingSystem.IsLinux() || !OperatingSystem.IsFreeBSD())
                 || !string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase))
             {
                 return null;
diff --git a/MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs b/MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs
index 89c195828..c42ec949b 100644
--- a/MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs
+++ b/MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs
@@ -332,7 +332,7 @@ namespace MediaBrowser.MediaEncoding.Encoder
 
         public bool CheckVaapiDeviceByDriverName(string driverName, string renderNodePath)
         {
-            if (!OperatingSystem.IsLinux())
+            if (!OperatingSystem.IsLinux() || !OperatingSystem.IsFreeBSD())
             {
                 return false;
             }
diff --git a/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs b/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs
index 539996c3a..9723b2957 100644
--- a/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs
+++ b/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs
@@ -161,7 +161,7 @@ namespace MediaBrowser.MediaEncoding.Encoder
                 _isPkeyPauseSupported = validator.CheckSupportedRuntimeKey("p      pause transcoding");
 
                 // Check the Vaapi device vendor
-                if (OperatingSystem.IsLinux()
+                if ((OperatingSystem.IsLinux() || OperatingSystem.IsFreeBSD())
                     && SupportsHwaccel("vaapi")
                     && !string.IsNullOrEmpty(options.VaapiDevice)
                     && string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase))
